test/unit/server/handlers/thirdpartyRequests/transactions/{ID}/authorizations.test.ts

```diff
@@ -0,0 +1,393 @@
+
+/*
+ * Mock Handler Functions
+ */
+const mockIsPayloadPending = jest.spyOn(Domain, 'isPayloadPending')
+const mockHasMatchingScope = jest.spyOn(Domain, 'hasMatchingScopeForPayload')
+const mockPutErrorRequest = jest.spyOn(Domain, 'putErrorRequest')
+const mockHasActiveCredential = jest.spyOn(
+  Domain,
+  'hasActiveCredentialForPayload'
+)
+
+const mockVerifySignature = jest.spyOn(Challenge, 'verifySignature')
+
+const mockRetrieveConsent = jest.spyOn(consentDB, 'retrieve')
+const mockRetrieveAllScopes = jest.spyOn(scopeDB, 'retrieveAll')
+
+const mockLoggerPush = jest.spyOn(Logger, 'push')
+const mockLoggerError = jest.spyOn(Logger, 'error')
+
+const mockPutThirdpartyTransactionsAuth = jest.spyOn(
+  thirdPartyRequest,
+  'putThirdpartyRequestsTransactionsAuthorizations'
+)
+
+const mockValidateAndVerifySignature = jest.spyOn(
+  Handler,
+  'validateAndVerifySignature'
+)
+
+/*
+ * Mock Request and Response Resources
+ */
+const payload: Domain.AuthPayload = {
+  consentId: '1234',
+  sourceAccountId: 'pisp-2343-f223',
+  status: 'PENDING',
+  challenge: 'QuoteResponse Object JSON string',
+  value: 'YjYyODNkOWUwZjUxNzOThmMjllYjE2Yg=='
+}
+
+// @ts-ignore
+const request: Request = {
+  headers: {
+    'fspiop-source': 'switch'
+  },
+  params: {
+    id: '1234'
+  },
+  payload: payload
+}
+
+// @ts-ignore
+const h: ResponseToolkit = {
+  response: (): ResponseObject => {
+    return {
+      code: (statusCode: number): ResponseObject => {
+        return {
+          statusCode
+        } as unknown as ResponseObject
+      }
+    } as unknown as ResponseObject
+  }
+}
+
+/*
+ * Mock consent and scopes
+ */
+const mockConsent: Consent = {
+  id: payload.consentId,
+  credentialStatus: 'ACTIVE',
+  credentialPayload: 'Mock public key payload'
+}
+
+const mockScopes: Scope[] = [
+  {
+    consentId: payload.consentId,
+    action: 'account.transfer',
+    accountId: payload.sourceAccountId
+  },
+  {
+    consentId: payload.consentId,
+    action: 'account.balance',
+    accountId: 'dfsp-2321-ahsh'
+  }
+]
+
+/*
+ * Incoming POST `/thirdpartyRequests/transaction/{ID}/authorizations'
+ * Async Handler Helper Unit Tests
+ */
+describe('validateAndVerifySignature', (): void => {
+  beforeEach((): void => {
+    // Positive flow values for a successful 202 return
+    mockIsPayloadPending.mockReturnValue(true)
+    mockHasActiveCredential.mockReturnValue(true)
+    mockHasMatchingScope.mockReturnValue(true)
+    mockVerifySignature.mockReturnValue(true)
+    // mockPutErrorRequest.mockImplementation(async (): Promise<void> => {})
+    mockPutErrorRequest.mockResolvedValue(undefined)
+
+    mockLoggerPush.mockReturnValue(null)
+    mockLoggerError.mockReturnValue(null)
+
+    mockRetrieveConsent.mockResolvedValue(mockConsent)
+    mockRetrieveAllScopes.mockResolvedValue(mockScopes)
+
+    mockPutThirdpartyTransactionsAuth.mockResolvedValue({
+      statusCode: 200,
+      headers: null,
+      data: Buffer.from('Response Data')
```

Comment:
why Buffer here? take a look at how it is implemented: 

https://github.com/mojaloop/sdk-standard-components/blob/master/src/lib/requests/baseRequests.js#L169

https://github.com/mojaloop/sdk-standard-components/blob/master/src/lib/request.js#L60

https://github.com/mojaloop/sdk-standard-components/blob/master/src/lib/requests/common.js#L90